nxdrive/gui/api.py

```diff
@@ -201,6 +202,34 @@ def get_transfers(self) -> List[Dict[str, Any]]:
 
         return result
 
+    def upload_is_direct_transfer(self, path, engine):
+        if engine.local_folder not in path.parents:
+            return True
+        elif not engine.dao.get_state_from_local(path.relative_to(engine.local_folder)):
+            return True
+        return False
+
+    @pyqtSlot(result=list)
+    def get_direct_transfers(self) -> List[Dict[str, Any]]:
```

Comment:
Missing doctstring + changelog.
```suggestion
    def get_direct_transfer_items(self) -> List[Dict[str, Any]]:
```